src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs

```diff
@@ -106,31 +106,9 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
 					matrix *= Matrix3x2.CreateScale(bounds.Width, bounds.Height);
 
 					// The image rescaling (i.e resampling) algorithm in the shader directly is really low quality (but really fast).
-					// SkiaSharp 3 introduces new SKSamplingOptions, but for now there is no alternative to writing
-					// the shader logic by hand.
-					// When we move to SkiaSharp3, remove this
-					// ******************************************************************************
-					if (_effect is null)
-					{
-						_effect = SKRuntimeEffect.Create(ImageResamplingShader, out var error);
-						if (error is { } e)
-						{
-							throw new InvalidOperationException(e);
-						}
-					}
-
-					var uniforms = new SKRuntimeEffectUniforms(_effect)
-					{
-						{ "imageSize", new[] { (float)backgroundArea.Width, (float)backgroundArea.Height } }
-					};
-					var children = new SKRuntimeEffectChildren(_effect)
-					{
-						["image"] = SKShader.CreateImage(scs.Image, SKShaderTileMode.Decal, SKShaderTileMode.Decal)
-					};
-
-					var imageShader = _effect.ToShader(false, uniforms, children, matrix.ToSKMatrix());
-					// ******************************************************************************
-					// and replace with this one line (and maybe add mipmapping
+					// There is no sound workaround for this at the moment. See https://github.com/unoplatform/uno/issues/17325.
+					var imageShader = SKShader.CreateImage(scs.Image, SKShaderTileMode.Decal, SKShaderTileMode.Decal, matrix.ToSKMatrix());
+					// SkiaSharp 3 introduces new SKSamplingOptions. When we move to SkiaSharp 3, replace the line about with this one.
```

Comment:
With SkiaSharp 3, we'll be able to have the good quality and also keeping everything working as intended?

Reply:
Yes.